modules/gapi/test/common/gapi_core_tests_inl.hpp

```diff
@@ -1377,6 +1386,169 @@ TEST_P(NormalizeTest, Test)
     }
 }
 
+TEST_P(KMeansNDNoInitTest, AccuracyTest)
+{
+    const int quantity = sz.height, dimensions = sz.width;
+    const cv::TermCriteria criteria(TermCriteria::MAX_ITER + TermCriteria::EPS, 30, 0);
+    const int attempts = 1;
+    double compact_gapi = -1.;
+    cv::Mat labels_gapi, centers_gapi;
+    // G-API code //////////////////////////////////////////////////////////////
+    cv::GMat in, labels, centers;
+    cv::GOpaque<double> compactness;
+    std::tie(compactness, labels, centers) = cv::gapi::kmeans(in, K, criteria, attempts, flags);
+    cv::GComputation c(cv::GIn(in), cv::GOut(compactness, labels, centers));
+    c.apply(cv::gin(in_mat1), cv::gout(compact_gapi, labels_gapi, centers_gapi), getCompileArgs());
+    // Validation //////////////////////////////////////////////////////////////
+    {
+        EXPECT_GE(compact_gapi, 0.);
+        EXPECT_EQ(labels_gapi.cols, 1);
+        EXPECT_EQ(labels_gapi.rows, quantity);
+        EXPECT_FALSE(labels_gapi.empty());
+        EXPECT_EQ(centers_gapi.cols, dimensions);
+        EXPECT_EQ(centers_gapi.rows, K);
+        EXPECT_FALSE(centers_gapi.empty());
+    }
```

Comment:
What about comparing results of gapi with ocv for input Mat(ND) case?

Reply:
It happens in the next test. This test is to check the functionality without `bestLabel` initialization.
Why doesn't this test compare with OCV? Because without initialization, `bestLabels` are chosen randomly by kmeans itself, so there are no possibility to compare